ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -372,7 +372,13 @@ public void onProgress(long bytesWritten, long contentLength, boolean done) {
           return;
         }
       } else {
-        requestBody = RequestBody.create(contentMediaType, body.getBytes(StandardCharsets.UTF_8));
+        // Use getBytes() to convert the body into a byte[], preventing okhttp from
+        // appending the character set to the Content-Type header when otherwise unspecified
+        // https://github.com/facebook/react-native/issues/8237
+        Charset charset = contentMediaType == null
+          ? StandardCharsets.UTF_8
+          : contentMediaType.charset(StandardCharsets.UTF_8);
+        requestBody = RequestBody.create(contentMediaType, body.getBytes(charset));
```

Comment:
@hey99xx I updated this line to include your suggestion for null safety. Just for posterity too, this matches how other parts of the code are handling this case:

https://github.com/facebook/react-native/blob/cda8171af30815edfa331e07d1bbf605f0926303/ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java#L562-L563